src/Elastic.Transport/Components/Pipeline/DefaultResponseBuilder.cs

```diff
@@ -314,17 +331,20 @@ private static bool SetSpecialTypes<TResponse>(string mimeType, byte[] bytes, St
 				{
 					["body"] = new DynamicValue(bytes.Utf8String())
 				};
-				cs = new DynamicResponse(dictionary) as TResponse;
+				response = new DynamicResponse(dictionary) as TResponse;
 			}
 			else
 			{
 				using var ms = memoryStreamFactory.Create(bytes);
 				var body = LowLevelRequestResponseSerializer.Instance.Deserialize<DynamicDictionary>(ms);
-				cs = new DynamicResponse(body) as TResponse;
+				response = new DynamicResponse(body) as TResponse;
 			}
 		}
 
-		return cs != null;
+		if (!response.LeaveOpen)
```

Comment:
++ agree with @flobernd, while it does no harm disposing it here as well it adds more complexity to this method.

Reply:
Yep, I will refactor. We already had a case in `SwapStreams` where we dispose of the original response stream, which should then release the connection as well. I don't think it matters if we hold the connection a fraction longer and strip that away, too.